data/templates/get_driveid.js

```diff
@@ -0,0 +1,196 @@
+            if (line.indexOf('pci') === 0) {
+                pciLines.push([line.slice(nameIndex + 1), line.slice(0,idIndex)]);
+            }
+        }
+    });
+
+    return pciLines.map(function(line) {
+        var scsiId = line[1].slice(line[1].lastIndexOf('-') + 1), vdStr;
+        if (scsiId.split(':').length === 4){
+		    var scsiIdArray = scsiId.split(':');
+            vdStr = ['/c', scsiIdArray[0], '/v', scsiIdArray[2]].join('');
+		}
+	    return [line[0], vdStr, scsiId];
+    });
+}
+
+/**
+* Parse scsi ID output via lsscsi command
+* @param {String} data
+*/
+function parseScsiInfo(lsscsiList) {
+	var lines = lsscsiList.split('\n');
+	return lines.map(function(line) {
+		if(line){
+		return [line.slice(line.lastIndexOf('/') + 1).replace(' ', ''),
+            line.slice(1, line.indexOf(']'))];
+		}
+	});
+}
+
+/**
+ * Build the drive mapping table
+ * @param {String} wwidData
+ * @param {String} vdData
+ * @return 0 if success, otherwise failed.
+ */
+function buildDriveMap(wwidData, vdData, scsiData) {
+    var parsedWwids = parseDriveWwid(wwidData),
+        scsiList = parseScsiInfo(scsiData),
+        vdList = parseVdInfo(vdData);
+    var linuxWwids = parsedWwids.linuxDriveIds, esxiWwids = parsedWwids.esxiDriveIds;
+    var driveIds=[];
+    esxiWwids.forEach(function(esxiWwid){
+        var vd = '', scsiId = '', diskPath = esxiWwid[0];
+        vdList.forEach(function(elem){
+            if (typeof elem !== 'undefined') {
+                if (elem[0] === diskPath) {
+                    vd = elem[1];
+                }
+            }
+        });
+        scsiList.forEach(function(elem){
+            if (typeof elem !== 'undefined'){
+                if(elem[0] === diskPath){
+                    scsiId = elem[1];
+                }
+            }
+        });
+        driveIds.push({"scsiId": scsiId, "virtualDisk": vd,
+            "esxiWwid": esxiWwid[1], "devName": diskPath});
+    });
+    var k=0;
+    linuxWwids.forEach(function(linuxWwid){
+        driveIds[k].identifier = k;
+        driveIds[k].linuxWwid = linuxWwid[1];
+        k += 1;
+    });
+    console.log(JSON.stringify(driveIds));
+    return 0;
+}
+
+function run() {
+    var wwidData, vdData, scsiData;
+    exec(cmdDriveWwid, options, function(err0, stdout0) {
+        if (err0) {
+            console.error(err0.toString());
+            process.exit(1);
+        }
+        wwidData = stdout0;
+        exec(cmdVdInfo, options, function(err1, stdout1) {
+            if (err1) {
+                console.error(err1.toString());
+                process.exit(1);
+            }
+            vdData = stdout1;
+            exec(cmdScsiId, options, function(err2, stdout2) {
+                if (err2) {
+                    console.error(err2.toString());
+                    process.exit(1);
+                }
+                scsiData = stdout2;
+                if (buildDriveMap(wwidData, vdData, scsiData)) {
```

Comment:
Should this be wrapped in a try/catch to fail gracefully in case the parsing fails, or am I missing parser error handling somewhere?